src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -205,249 +170,45 @@ where
         ReclockFollower::new(as_of)
     };
 
-    let (
-        SourceConnectionStreams {
-            batches,
-            batch_upper_summaries,
-            health_stream,
-        },
-        source_reader_token,
-    ) = source_reader_operator(
-        scope,
-        config.clone(),
-        source_connection,
-        connection_context,
-        reclock_follower.share(),
-        &resume_stream,
-    );
-
-    let (remap_stream, remap_token) =
-        remap_operator(scope, config.clone(), batch_upper_summaries, &resume_stream);
-
-    let ((reclocked_stream, reclocked_err_stream), _reclock_token) = reclock_operator(
-        scope,
-        config.clone(),
-        reclock_follower,
-        batches,
-        remap_stream,
-    );
-
-    let health_token = health_operator(scope, config, health_stream);
-
-    let token = Rc::new((source_reader_token, remap_token, resume_token, health_token));
+    let (resume_tx, resume_rx) = tokio::sync::mpsc::unbounded_channel();
+    let (source_tx, source_rx) = tokio::sync::mpsc::unbounded_channel();
+    let (source_upper_tx, source_upper_rx) = tokio::sync::mpsc::unbounded_channel();
+
+    resume_stream.capture_into(UnboundedTokioCapture(resume_tx));
+    let reclocked_resume_stream =
+        reclock_resume_upper(resume_rx, reclock_follower.share(), worker_id, id);
```

Comment:
`*_stream` values kindof mean timely streams in this function, but this is an async stream, maybe we can come up with a better name?